lib/doorkeeper/oauth/scopes.rb

```diff
@@ -70,10 +70,26 @@ def <=>(other)
         end
       end
 
+      # DEPRECATED: With dynamic scopes, #allowed should be called because
+      # A & B doesn't really make sense with dynamic scopes.
+      #
+      # For example, if A = user:* and B is user:1, A & B = [].
+      # If we modified this method to take dynamic scopes into an account, then order
+      # becomes important, and this would violate the principle that A & B = B & A.
       def &(other)
         self.class.from_array(all & to_array(other))
```

Comment:
Can we somehow check here if dynamic scopes used? And run `allowed` behavior? Wanna to avoid braking changes in case somebody uses scores directly

Reply:
@nbulaj Ok, updated!